modules/api/src/main/java/org/apache/ignite/app/Ignite.java

```diff
@@ -29,4 +30,11 @@
      * @return Table manager.
      */
     TableManager tableManager();
+
+    /**
+     * Gets interactive object for configure cluster.
+     *
+     * @return Ignite configuration.
+     */
+    ConfigurationRegistry configuration();
```

Comment:
It's not valid to use ConfigurationRegistry as public API, cause it's located in inner configuration module. Semantically we need some new  ConfigurationRegistry interface, locates in ignite-api module, that will have only subset of current ConfigurationRegistry methods, for example we don't need startStorageConfigurations(), constructors, etc as part of public API. It also means that we should rename ConfigurationRegistry to ConfigurationRegistryImpl or similar.